### PR TITLE
Add compat data for ServiceWorker

### PR DESCRIPTION
**Pull request by bunnybooboo**
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker

as per #1002 notes on ESR might need editing

### CODE REVIEW

File: api/ServiceWorker.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            "version_added": "40"
+          },
+          "chrome": {
+            "version_added": "40"
+          },
+          "chrome_android": {
+            "version_added": "40"
+          },
+          "edge": {
+            "version_added": "17"
```
Comment (teoli2003) [inline api/ServiceWorker.json]: "flags" (the other SW entries have it – why not this one?)

File: api/ServiceWorker.json [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+              "version_added": "40"
+            },
+            "chrome": {
+              "version_added": "40"
+            },
+            "chrome_android": {
+              "version_added": "40"
+            },
+            "edge": {
+              "version_added": "17"
```
Comment (teoli2003) [inline api/ServiceWorker.json]: "flags"

File: api/ServiceWorker.json [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (teoli2003) [inline api/ServiceWorker.json]: "flags"

File: api/ServiceWorker.json [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (teoli2003) [inline api/ServiceWorker.json]: "flags"

**Comment by bunnybooboo** [comment]
On consulting [Edge Dev API catalog](https://developer.microsoft.com/en-us/microsoft-edge/platform/catalog/?page=1&q=serviceworker), [and](https://steffenjahr.de/2017/05/12/microsoft-edge-and-you-pwa-service-worker-ahead/) [several](https://blogs.windows.com/msedgedev/2017/10/17/edgehtml-16-fall-creators-update/) [blogs](https://www.infoworld.com/article/3225459/web-applications/progressive-web-apps-in-microsoft-edge-what-you-need-to-know.html), I'm lead to believe the preview is enabled only via the pref. Version 17 will have service workers [enabled by default](https://blogs.windows.com/msedgedev/2018/02/06/welcoming-progressive-web-apps-edge-windows-10/).

**Comment by bunnybooboo** [comment]
Further discussion in #1013 this should finally be up to scratch. :crossed_fingers:

**Comment by bunnybooboo** [comment]
Via discussion w/ Samsung dev advocate [on Mastodon](https://toot.cafe/@peter/99517672960971594) added Samsung Internet data.

**Pull request merged**